test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -5981,15 +5981,31 @@ def submit(self, obj, script=None, extend=None, submit_dir=None,
             os.chdir(submit_dir)
         c = None
         # 1- Submission using the command line tools
+        runcmd = []
+        exp_str = ""
+        if len(env_set) > 0:
+            if '()' in list(env_set.keys())[0]:
+                func = list(env_set.keys())[0]
+                f_body = list(env_set.values())[0]
+                f_name = func.replace('()', '')
+                runcmd += [func, f_body, "\n", "export", "-f", f_name]
```

Comment:
I'm not sure I fully understand this.  This only checks the first element of the env_set if it is a function.  What if a further element is?